default scheme (#27310)
dotnet/AspNetCore.Docs: aspnetcore/security/authentication/index.md

Authentication is responsible for providing the <xref:System.Security.Claims.ClaimsPrincipal> for authorization to make permission decisions against. There are multiple authentication scheme approaches to select which authentication handler is responsible for generating the correct set of claims:

* [Authentication scheme](#authentication-scheme)
* The default authentication scheme, discussed in the next two sections.
* Directly set <xref:Microsoft.AspNetCore.Http.HttpContext.User?displayProperty=nameWithType>.

When there is only a single authentication scheme registered, it becomes the default scheme. If multiple schemes are registered and the default scheme isn't specified, a scheme must be specified in the authorize attribute, otherwise, the following error is thrown:

> InvalidOperationException: No authenticationScheme was specified, and there was no DefaultAuthenticateScheme found. The default schemes can be set using either AddAuthentication(string defaultScheme) or AddAuthentication(Action\<AuthenticationOptions> configureOptions).

### `DefaultScheme`

When there is only a single authentication scheme registered, the single authentication scheme:

> * Is automatically is used as the `DefaultScheme`.

Reviewer: ```suggestion
* Is automatically used as the `DefaultScheme`.
```

> * Eliminates the need to specify the `DefaultScheme` in `AddAuthentication`.

Reviewer 2: maybe also mention via `AuthenticationOptions` to?

Author: > maybe also mention via `AuthenticationOptions` too?

@HaoK  can you check [this commit](https://github.com/dotnet/AspNetCore.Docs/pull/27310/commits/b4196a448fa1776480ede3c27579fe67c02fd3d5)

> * Eliminates the need to specify the `DefaultScheme` in `AddAuthentication` or [`AddAuthentication(IServiceCollection, Action<AuthenticationOptions>)`](/dotnet/api/microsoft.extensions.dependencyinjection.authenticationservicecollectionextensions.addauthentication#microsoft-extensions-dependencyinjection-authenticationservicecollectionextensions-addauthentication(microsoft-extensions-dependencyinjection-iservicecollection-system-action((microsoft-aspnetcore-authentication-authenticationoptions))))

Reviewer: ```suggestion
* Eliminates the need to specify the `DefaultScheme` in `AddAuthentication` or <xref:Microsoft.Extensions.DependencyInjection.AuthenticationCoreServiceCollectionExtensions.AddAuthenticationCore(Microsoft.Extensions.DependencyInjection.IServiceCollection,System.Action{Microsoft.AspNetCore.Authentication.AuthenticationOptions})>.
```

> The single authentication scheme being automatically used as the `DefaultScheme` can be disabled via `AppContext.SetSwitch("Microsoft.AspNetCore.Authentication.SuppressAutoDefaultScheme")`.

Reviewer: ```suggestion
To disable automatically using the single authentication scheme as the `DefaultScheme`, call `AppContext.SetSwitch("Microsoft.AspNetCore.Authentication.SuppressAutoDefaultScheme")`.
```
